lessons/python/inttopics/inttopics.ipynb

```diff
@@ -0,0 +1,672 @@
+    "for i in range(len(mylist)):\n",
+    "    if mylist[i]%2 == 0:\n",
+    "        break\n",
+    "    print(mylist[i]) "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# continue will immediately jump to the next iteration\n",
+    "for i in range(len(mylist)):\n",
+    "    if mylist[i]%2 == 0:\n",
+    "        continue\n",
+    "    print(mylist[i]) "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "mylist.append('hello')\n",
+    "print(mylist) "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# pass is useful to make certain conditions explicit and take care of outside cases\n",
+    "# while it's not 'necessary' here for the loop to function, it makes the operation clearer\n",
+    "\n",
+    "for i in range(len(mylist)):\n",
+    "    if type(mylist[i]) is int:\n",
+    "        mylist[i] = mylist[i] * 0.5\n",
+    "    if type(mylist[i]) is str:\n",
+    "        pass\n",
+    "    \n",
+    "print(mylist)      "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Assertions ('Sanity Checks')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's say the following function was in the middle of a long script that runs input data through a series of operations."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def sumdoubles(x,y):\n",
+    "    '''Multiplies inputs by 2, and returns their sum.'''\n",
+    "    out = x*2 + y*2\n",
+    "    return(out) "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "sumdoubles(2,3) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "What happens if something goes wrong, and a pair of strings are somehow given as input?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "sumdoubles('hello', 'how goes?') "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "That output could be really bad for the downstream parts of our script if they're expecting numerical input!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# assertions are handy to avoid the propagation of errors\n",
+    "\n",
+    "def sumdoubles(x,y):\n",
+    "    assert(type(x) and type(y) != str)\n",
```

Comment:
This logical expression is incorrect; you're asserting that `type(x)` is truthy and `type(y) != str`, not that both types are `str`. Should be `assert not isinstance(x, str) and not isinstance(y, str)` (parentheses unnecessary.)